builtin/logical/mysql/secret_creds.go

```diff
@@ -90,23 +90,31 @@ func (b *backend) secretCredsRevoke(
 	if !ok {
 		roleName = ""
 	} else {
-		roleName, ok = roleNameRaw.(string)
+		roleName, _ = roleNameRaw.(string)
 	}
 	// init default revoke sql string.
 	// this will replaced by a user provided one if one exists
 	// otherwise this is what will be used when lease is revoked
 	revokeSQL := defaultRevokeSQL
 
+	// init bool to track if we should responding with warning about nil role
```

Comment:
You want to add the warning at the point that it's created. Don't use a bool; instead, create a nil logical.Response pointer above. When the warning should be created, check if it's still nil; if so assign the pointer to a new response. Then in either case, add the warning. At the end of the function, return the response, which will either be nil or contain warnings.